packages/tendermint-rpc/src/tendermint34/adaptors/v0-34/responses.ts

```diff
@@ -416,17 +416,23 @@ type RpcSignature = {
   readonly signature: string | null;
 };
 
+/**
+ * In some cases a timestamp is optional and set to the value 0 in Go.
+ * This can lead to strings like "0001-01-01T00:00:00Z" (see https://github.com/cosmos/cosmjs/issues/704#issuecomment-797122415).
+ * This decoder tries to clean up such encoding from the API and turn them
+ * into undefined values.
+ */
+function decodeOptionalTime(timestamp: string): DateWithNanoseconds | undefined {
```

Comment:
Nice this looks clean.

You only get these absent commit signatures in a multi-node environment, when one node is down (or most recent commit when one has slow network)